internal/auth.go

```diff
@@ -196,6 +200,21 @@ func ClearCSRFCookie(r *http.Request) *http.Cookie {
 	}
 }
 
+// FindCSRFCookie extracts the CSRF cookie from the request based on state.
+func FindCSRFCookie(r *http.Request) (c *http.Cookie, err error) {
+	state := r.URL.Query().Get("state")
+	if len(state) < 34 {
+		return nil, errors.New("Invalid CSRF state value")
+	}
```

Comment:
Please could we move the state extraction into `AuthCallbackHandler` and then move this check into a new `ValidateState` function? Then it can be removed from here and `ValidateCSRFCookie`

Reply:
Done (at least to my understanding what you meant).